snap-server/src/main/java/com/kyc/snap/words/WordsearchSolver.java

```diff
@@ -45,30 +51,33 @@
                                         && col < width; row += di, col += dj) {
                                     word += grid.get(row).charAt(col);
                                     positions.add(new Point(col, row));
-                                    if (word.length() >= 2 && wordFrequencies.containsKey(word))
-                                        results.add(new Result(word, new ArrayList<>(positions)));
+                                    if (word.length() >= 2 && words.contains(word))
+                                        results.add(new Result(word, new ArrayList<>(positions), wordbankSet.contains(word)));
                                 }
                                 positions.clear();
                             }
         }
         return results.stream()
-                .sorted(Comparator.comparing(result -> -Math.sqrt(wordFrequencies.get(result.word)) * Math.pow(26, result.word.length())))
+                .sorted(Comparator.comparing(result -> (
+                		result.inWordbank ? (WORDBANK_WEIGHT * result.word.length()) : 
+                			-Math.sqrt(wordFrequencies.get(result.word)) * Math.pow(26, result.word.length())
+                		)))
```

Comment:
Let's make the fact that word bank words should be ordered first more explicit
```suggestion
                .sorted(Comparator.<Result, Boolean>comparing(result -> !result.inWordbank)
                    .thenComparing(result -> -Math.sqrt(wordFrequencies.getOrDefault(result.word, 1L)) * Math.pow(26, result.word.length())))
```
and remove `WORDBANK_WEIGHT` above.

This will order word bank words that are also in the dictionary higher, but I think that's fine. Otherwise could also tack on a `result.inWordbank ? -result.word.length : -Math.sqrt(...`.

Reply:
Cool didn't know that was possible. I tacked on the extra logic because we'd always want the longer wordbank phrase to have priority in highlighting, in case are any words are substrings of each other e.g. "WALUIGI" >> "LUIGI"